src/coco/repository/util/repositoryUtil.ts

```diff
@@ -0,0 +1,86 @@
+import RepositoryType from "../model/RepositoryType";
+import Repository from "../model/Repository";
+import { execSync } from "child_process";
+import Logger from "../../../log/Logger";
+
+const GET_GIT_REPOSITORY_COMMAND = "git config --local --get remote.origin.url";
+const GIT_REPO_URL_SUFFIX = ".git";
+const DOMAIN_GITHUB = "github.com";
+const DOMAIN_GITLAB = "gitlab.com";
+const DOMAIN_BITBUCKET = "bitbucket.org";
+const CUSTOM_DOMAIN_REG_EXP_GITHUB = /github.(.+).com/; // TODO
+const SUFFIX_HTTPS = "https://";
+
+function getRepositoryForType(rootFolderPath: string, type: RepositoryType): Repository {
+    let fullUrl: string | undefined;
+    try {
+        fullUrl = execSync(GET_GIT_REPOSITORY_COMMAND, {
+            cwd: rootFolderPath
+        }).toString().trim();
+    } catch (error) {
+        Logger.error(`Getting git repository url failed`, error);
+    }
+
+    if (!fullUrl) {
+        return getDefaultRepository();
+    }
+
+    if (fullUrl.endsWith(GIT_REPO_URL_SUFFIX)) {
+        fullUrl = fullUrl.substring(0, fullUrl.length - GIT_REPO_URL_SUFFIX.length);
+    }
+
+    const defaultUrl = getDefaultUrlForType(type);
+    const defaultUrlIndex = fullUrl.indexOf(defaultUrl);
+    if (defaultUrlIndex >= 0) {
+        const name = fullUrl.substring(defaultUrlIndex + defaultUrl.length + 1); // +1 is for the divider after domain
+        return {
+            repository: name
+        };
+    }
+
+    const customDomainRegExp = getCustomDomainRegExpForType(type);
+    const customUrl = customDomainRegExp?.exec(fullUrl)?.[0];
+    if (customUrl) {
+        const customUrlIndex = fullUrl.indexOf(customUrl);
+        const name = fullUrl.substring(customUrlIndex + customUrl.length + 1); // +1 is for the divider after domain
+        const fullCustomUrl = SUFFIX_HTTPS + customUrl;
```

Comment:
We cannot just strip the original protocol and force https. For instance, locally hosted repos might not be SSL enabled, and this would fail.

Reply:
i think we can just guess here, the input we have is like below when ssh is used
git@github.com:zeplin/vscode-extension

Just a note: This part is just an autofill, so if the data is wrong or could not be parsed, the user can always enter is manually

Reply:
We can split the repo origin parsing and get the correct result?

Reply:
I didn't understand that. How?

Reply:
`url.startsWith("http")` :vs: `url.startsWith("git")` or regex equivalent.

Reply:
first one is ok, for second one, we should still assume it is https, right?

Reply:
done in https://github.com/zeplin/vscode-extension/pull/14
if http/https is not known https is assumed.